warehouse/jobs/runner.go

```diff
@@ -35,75 +36,52 @@ func InitWarehouseJobsAPI(
 }
 
 func WithConfig(a *AsyncJobWh, config *config.Config) {
-	a.MaxBatchSizeToProcess = config.GetInt("Warehouse.jobs.maxBatchSizeToProcess", 10)
-	a.MaxCleanUpRetries = config.GetInt("Warehouse.jobs.maxCleanUpRetries", 5)
-	a.MaxQueryRetries = config.GetInt("Warehouse.jobs.maxQueryRetries", 3)
-	a.MaxAttemptsPerJob = config.GetInt("Warehouse.jobs.maxAttemptsPerJob", 3)
-	a.RetryTimeInterval = config.GetDuration("Warehouse.jobs.retryTimeInterval", 10, time.Second)
-	a.AsyncJobTimeOut = config.GetDuration("Warehouse.jobs.asyncJobTimeOut", 300, time.Second)
+	a.maxBatchSizeToProcess = config.GetInt("Warehouse.jobs.maxBatchSizeToProcess", 10)
+	a.maxCleanUpRetries = config.GetInt("Warehouse.jobs.maxCleanUpRetries", 5)
+	a.maxQueryRetries = config.GetInt("Warehouse.jobs.maxQueryRetries", 3)
+	a.maxAttemptsPerJob = config.GetInt("Warehouse.jobs.maxAttemptsPerJob", 3)
+	a.retryTimeInterval = config.GetDuration("Warehouse.jobs.retryTimeInterval", 10, time.Second)
+	a.asyncJobTimeOut = config.GetDuration("Warehouse.jobs.asyncJobTimeOut", 300, time.Second)
 }
 
 func (a *AsyncJobWh) getTableNamesBy(sourceID, destinationID, jobRunID, taskRunID string) ([]string, error) {
 	a.logger.Infof("[WH-Jobs]: Extracting table names for the job run id %s", jobRunID)
 	var tableNames []string
 	var err error
-	query := `SELECT id
-		FROM
-		` + warehouseutils.WarehouseUploadsTable + `
-		WHERE metadata->>'source_job_run_id'=$1
-			AND metadata->>'source_task_run_id'=$2
-			AND metadata in (
-				SELECT
-					metadata
-				FROM
-				` + warehouseutils.WarehouseUploadsTable + `
-				WHERE source_id=$3
-				AND destination_id=$4
-			)`
+
+	query := `SELECT table_name FROM ` + warehouseutils.WarehouseTableUploadsTable + ` WHERE wh_upload_id IN ` +
+		` (SELECT id FROM ` + warehouseutils.WarehouseUploadsTable + ` WHERE metadata->>'source_job_run_id'=$1
+				AND metadata->>'source_task_run_id'=$2
+				AND source_id=$3
+				AND destination_id=$4)`
 	a.logger.Debugf("[WH-Jobs]: Query is %s\n", query)
-	rows, err := a.dbHandle.Query(query, jobRunID, taskRunID, sourceID, destinationID)
+	rows, err := a.dbHandle.QueryContext(a.context, query, jobRunID, taskRunID, sourceID, destinationID)
 	if err != nil {
-		a.logger.Errorf("[WH-Jobs]: Error carrying out the query %s ", query)
-		return nil, err
+		a.logger.Errorf("[WH-Jobs]: Error executing the query %s with error %s", query, err.Error())
```

Comment:
No need for `err.Error()`. Alternatively `%v` can be used.

https://go.dev/play/p/B493mQglal_g